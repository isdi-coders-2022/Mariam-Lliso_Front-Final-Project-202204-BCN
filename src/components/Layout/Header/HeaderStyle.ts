import styled from "styled-components";
import theme from "../../../theme/theme";


const HeaderStyle = styled.header`
  padding: 1.5rem clamp(1rem, 5%, 3rem);
  background: rgba( 255, 255, 255, 0.8);
  box-shadow: 0 4px 4px 0 rgb(0 0 0 / 10%);
  backdrop-filter: blur( 3.5px );
  border-radius: 0 0 10px 10px;

  .header {
    background-color: ${theme.palette.grey[100]};
    padding: 1rem clamp(1rem, 5%, 3rem);
    border-radius: 10px;
    box-shadow: 0px 4px 2px 0px rgb(0 0 0 / 5%);
    color: ${theme.palette.grey[900]}
  }

  .header__title {
      font-size: 20px
  }
`;

export default HeaderStyle;
