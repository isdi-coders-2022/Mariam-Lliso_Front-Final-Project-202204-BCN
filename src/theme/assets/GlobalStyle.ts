import { createGlobalStyle } from "styled-components";

const GlobalStyle = createGlobalStyle`
  h1 {
    margin:0;
  }
`;

export default GlobalStyle;
