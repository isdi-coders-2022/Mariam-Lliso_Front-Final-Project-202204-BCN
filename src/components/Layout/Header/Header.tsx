import { Box, Typography } from "@mui/material";
import HeaderStyle from "./HeaderStyle";

const Header = (): JSX.Element => {
  return (
    <HeaderStyle>
      <Box className="header">
        <Typography variant="h3" component="h1">
          Valencia Sin Gluten
        </Typography>
      </Box>
    </HeaderStyle>
  )
}

export default Header;
