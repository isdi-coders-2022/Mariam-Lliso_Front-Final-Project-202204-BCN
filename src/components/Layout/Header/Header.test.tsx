import { CssBaseline, ThemeProvider } from "@mui/material";
import GlobalStyle from "../../../theme/assets/GlobalStyle";
import theme from "../../../theme/theme";
import Header from "./Header";
import TestRenderer from "react-test-renderer";

describe("Given a RegisterPage component page", () => {
  describe("When it's invoked", () => {
    test("Then it should always match this snapshot", () => {
      const testedRegisterPage = TestRenderer.create(
      <ThemeProvider theme={theme}>
      <GlobalStyle />
      <CssBaseline />
      <Header />
      </ThemeProvider>
      );

      expect(testedRegisterPage).toMatchSnapshot();
    });
  });
});
