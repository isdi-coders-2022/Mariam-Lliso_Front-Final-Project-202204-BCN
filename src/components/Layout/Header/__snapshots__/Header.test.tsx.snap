// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a RegisterPage component page When it's invoked Then it should always match this snapshot 1`] = `
<header
  className="sc-bczRLJ jPsffr"
>
  <div
    className="header MuiBox-root css-0"
  >
    <h1
      className="MuiTypography-root MuiTypography-h3 css-7c3av3-MuiTypography-root"
    >
      Valencia Sin Gluten
    </h1>
  </div>
</header>
`;
